File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/idea/settings/interpreter/UpdateInterpreterHandler.scala
package org.intellij.plugin.zeppelin.idea.settings.interpreter

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.openapi.project.Project
import org.intellij.plugin.zeppelin.constants.ZeppelinConstants

/**
  * Updating after changing interpreters
  */
trait UpdateInterpreterHandler {
  def updateInterpreter(project: Project)
}

object UpdateInterpreterHandler {
  val EP_NAME: ExtensionPointName[UpdateInterpreterHandler] =
    ExtensionPointName.create[UpdateInterpreterHandler](ZeppelinConstants.PLUGIN_ID + ".updateInterpreterHandler")

  def getAll: Array[UpdateInterpreterHandler] = EP_NAME.getExtensions
}

File: zeppelin/src/org/intellij/plugin/zeppelin/extensionpoints/UpdateInterpreterHandler.kt
package org.intellij.plugin.zeppelin.extensionpoints

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.openapi.project.Project
import org.intellij.plugin.zeppelin.constants.ZeppelinConstants

/**
 * Updating after changing interpreters
 */
interface UpdateInterpreterHandler {
    fun updateInterpreter(project: Project)

    companion object {
        private val EP_NAME: ExtensionPointName<UpdateInterpreterHandler> = ExtensionPointName.create<UpdateInterpreterHandler>(
                ZeppelinConstants.PLUGIN_ID + ".updateInterpreterHandler")

        fun getAll(): Array<out UpdateInterpreterHandler> = EP_NAME.extensions
    }
}